compiler/dcalc/ast.mli

```diff
@@ -169,14 +204,12 @@ val make_let_in :
   Pos.t ->
   expr Pos.marked Bindlib.box
 
-val make_multiple_let_in :
-  Var.t array ->
-  typ Pos.marked list ->
-  expr Pos.marked Bindlib.box list ->
-  expr Pos.marked Bindlib.box ->
-  Pos.t ->
-  expr Pos.marked Bindlib.box
+(** {1 AST manipulation helpers}*)
 
-type binder = (expr, expr Pos.marked) Bindlib.binder
+val build_whole_scope_expr : decl_ctx -> scope_body -> Pos.t -> expr Pos.marked Bindlib.box
+(** Usage: [build_whole_scope_expr ctx body scope_position] where [scope_position] corresponds to
+    the line of the scope declaration for instance. *)
 
-type program = { decl_ctx : decl_ctx; scopes : (ScopeName.t * Var.t * expr Pos.marked) list }
+val build_whole_program_expr : program -> ScopeName.t -> expr Pos.marked Bindlib.box
```

Comment:
Why does the result is boxed ? It seems everywhere this function is used, its results is immediately unboxed.